### ISSUE TITLE
Does it support flight factor 320U?

### ISSUE THREAD

**Issue by demonguy** [jarangmandi/BFSGSimCom#51]
I happened to see that, it works on zibo 738 but not on Flight Factor 320.
Do you have plan to fix it?

**Comment by jarangmandi** [comment]
Hi demonguy - can you be a bit more specific about what's not working please?

Many third party aircraft create their own internal variables for management of aircraft systems that is more detailed that X-Plane provides by default. Sometimes that can be worked around, other times it can't!

From experience with the FFA320 though (it's one of the few third-party aircraft I have), the likely solution will be an edit to the 'published.txt' file and the XPUIPC configuration file which provides the mechanism to convert visible X-Plane datarefs to the FSUIPC offsets that BFSGSimCom uses.

In short, I may be able to provide a workaround to get it working. A coded fix will be less likely.

**Comment by demonguy** [comment]
Well. I've just done some tested:

1. with the latest version of teamspeak 3 and BFGSimCom which i just downloaded today. Nethier zibo 738 nor ff320 can change automatically. When enable detailed information. It shows all 4 frequencies as 0.0

2. with Teamspeak 3.1.6 and BFSGSimCom 0.11.1. It works on my zibo 738. However on ff320, it shows the correct frequencies, but with message "unrecognised radio selected" (I turned off 2nd VHF)

Is there anything else i can provide?

**Comment by jarangmandi** [comment]
That should be enough to be going on with for now... I'll focus on the first combination above and comment back here when I know what's going on.

**Comment by demonguy** [comment]
BTW. this is where is downloaded BFSGSimCom
https://www.myteamspeak.com/addons/0e1b55d0-f334-4c92-bdc5-e390cb2be4d0

**Comment by demonguy** [comment]
just menetion, i'm using xplane 14.1 with XPUIPC 2.0.38

**Comment by jarangmandi** [comment]
Hi - sorry it's taken a while.

You have downloaded the latest version of BFSGSimCom and looks like you're using the same version of X-Plane and  XPUIPC as I am, so hopefully this will be easy!

In version 0.13 of the plugin, the code was changed to take advantage of the ability of X-Plane to work with the 8.33 kHz spacing that is becoming more common. Unfortunately the default behaviou of XPUIPC doesn't consider this so a small edit to the XPUIPCOffsets.CFG is required.

Simply paste the following lines to the end of the file, and the COM1 and 2 radios in any X-Plane aircraft should start working again.

```
#
# Added for XP 8.33Khz processing.
#
Dataref	com1_freq_833	sim/cockpit2/radios/actuators/com1_frequency_hz_833	int
Dataref	com2_freq_833	sim/cockpit2/radios/actuators/com2_frequency_hz_833	int
Dataref	com1_stby_freq_833	sim/cockpit2/radios/actuators/com1_standby_frequency_hz_833	int
Dataref	com2_stby_freq_833	sim/cockpit2/radios/actuators/com2_standby_frequency_hz_833	int
Offset	0x05c4	UINT32	3	r	$com1_freq_833 1000 *
Offset	0x05c8	UINT32	3	r	$com2_freq_833 1000 *
Offset	0x05cc	UINT32	3	r	$com1_stby_freq_833 1000 *
Offset	0x05d0	UINT32	3	r	$com2_stby_freq_833 1000 *
```

Let me know how you get on!

**Comment by demonguy** [comment]
It works on zibo 738 now.
However ff320 still doesn't work. it shows "Unrecognised / More than one radio selected.". However i shut down com2 and com3. it still shows like this.

![image](https://user-images.githubusercontent.com/1451096/77317719-aa44df00-6d46-11ea-9dec-80c5b11d3ea1.png)

**Comment by jarangmandi** [comment]
I've tried this on the A320 sim tonight, and you're right in that when the Audio Control Panel (the panel in the bottom left of the pedestal in your picture) is set up the way you have it, the program reports that more than one radio is selected.

I have managed to reliably reproduce the issue that you're seeing - the plugin not picking up the selected frequency, and not responding to changes in the selected radios - and will issue a fix when I've found the time to code it.

The quick workaround is simple - you have two lit knobs on the left of the Audio Control Panel. One for VHF1 and one for VHF2. Make sure that only one of these is lit when you start TeamSpeak. The lit knob is the radio that's selected. Alternatively, change the BFSGSimCom mode to "Semi-Automatic" and place yourself in the root room (122.800 in your photo above) manually. This is the behaviour that I'll try to fix.

Note that in the real aircraft, it is the buttons above the switches that should define which radio is selected for transmit - only one of these can be lit at a time. Unfortunately, the default behaviour of the FFA320 is to populate the data that BFSGSimCom uses from the status of the knobs and not the switches above them, and it'll take a lot more to work out how to get past that.

**Comment by demonguy** [comment]
That's great. Thanks for your help
